agent/agent-bootstrap/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/rpc/MetricsView.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.rpc;
+
+import static io.opentelemetry.instrumentation.api.instrumenter.BootstrapSemanticAttributes.IS_SYNTHETIC;
+import static io.opentelemetry.instrumentation.api.instrumenter.BootstrapSemanticAttributes.TARGET;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+// this is temporary, see
+// https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3962#issuecomment-906606325
+@SuppressWarnings("rawtypes")
+final class MetricsView {
+
+  private static final Set<AttributeKey> alwaysInclude = buildAlwaysInclude();
+  private static final Set<AttributeKey> clientView = buildClientView();
+  private static final Set<AttributeKey> clientFallbackView = buildClientFallbackView();
+  private static final Set<AttributeKey> serverView = buildServerView();
+  private static final Set<AttributeKey> serverFallbackView = buildServerFallbackView();
+
+  private static Set<AttributeKey> buildAlwaysInclude() {
+    // the list of recommended metrics attributes is from
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/rpc.md#attributes
+    Set<AttributeKey> view = new HashSet<>();
+    view.add(SemanticAttributes.RPC_SYSTEM);
+    view.add(SemanticAttributes.RPC_SERVICE);
+    view.add(SemanticAttributes.RPC_METHOD);
+
+    // START APPLICATION INSIGHTS CODE
+    view.add(TARGET);
+    view.add(IS_SYNTHETIC);
+    // END APPLICATION INSIGHTS CODE
```

Comment:
👍 

(did we need the http metrics view for something similar?)

Reply:
no

Reply:
oh I see why it's not needed in http metrics, I think we can do the same in rpc metrics (call the view first, and then add our properties to the result) so that we can remove this `MetricsView` copy

Reply:
not sure i understand, let's chat.